vm/interpreter/src/default_runtime.rs

```diff
@@ -985,26 +985,35 @@ where
     ) -> Result<HashMap<Address, Vec<bool>>, Box<dyn StdError>> {
         // Gas charged for batch verify in actor
 
-        let out = vis
-            .par_iter()
-            .map(|(&addr, seals)| {
-                let results = seals
-                    .par_iter()
-                    .map(|s| {
-                        if let Err(err) = V::verify_seal(s) {
-                            debug!(
-                                "seal verify in batch failed (miner: {}) (err: {})",
-                                addr, err
-                            );
-                            false
-                        } else {
-                            true
-                        }
-                    })
-                    .collect();
-                (addr, results)
-            })
-            .collect();
+        let cpus = num_cpus::get();
+        // Create a local threadpool just for running verify seal in this epoch
+        let batch_pool = rayon::ThreadPoolBuilder::new()
```

Comment:
Is the Runtime object a singleton in the process? If so, I would create the threadpool in the constructor and store the object as a property on the Runtime.

Reply:
Mmm, like that could be done, but its kind of annoying because this Runtime trait is shared between all versions of actors. And thus we would need to change the interface, release a breaking version of the runtime, then release like 3 different version of actors too. I can add a todo and an associated issue for doing this in v5?

Reply:
Why does this change the interface? The threadpool should be internal to the Runtime object when you instantiate it in the constructor.

Reply:
Oh i misunderstood what you were saying. Let me try that out. It should work

Reply:
Actually, that wont really optimize anything. In fact, i think that adds even more overhead. A `Runtime` is instantiated every time we make a `Interpreter::send()` call. So if we were to make a threadpool in the `Runtime` constructor, we will be making it even when we don't use it (which will be 99%) of the time.  
This is because batch_verify_seals only gets called once at the end of the epoch by the cron actor. So currently we create a new threadpool at every epoch. What you are suggesting will create a threadpool for every transaction sent.

Reply:
Gotcha, this is why I was wondering if it is a singleton. Thanks for checking